pandas/tests/indexes/test_multi.py

```diff
@@ -766,6 +766,14 @@ def test_from_arrays_empty(self):
                 ValueError, "Must pass non-zero number of levels/labels"):
             MultiIndex.from_arrays(arrays=[])
 
+        # 0 levels, names defined
+        result = MultiIndex.from_arrays(arrays=[],
```

Comment:
The reasoning for requiring `[[], []]` in `from_arrays` is that `[]` has the wrong shape for the input, as opposed to in the case of `from_tuples`, where `[]` is the correct shape of what it expects as input.

But OK, it is a bit subjective, so I am fine with either.